### DISCUSSION TITLE
Modeling Dell MX7000 and associated IOM cards

### DISCUSSION THREAD

**Discussion by zeridon** [netbox-community/devicetype-library#2472] [General]
I am currently trying to model (faithfully) MX7000 chassis and it's various IOM's/etc.

Netbox itself is used for both rack, device, network information, so we need to model a rack with a chassis. Inside the chassy blades and IOM cards. On the dc/location level we need to track cabling between switches and the IOM's. On the device level we need to track both chassis, blade, iom. On the network level we need to track interfaces (on the IOM's? on the Blades?) addressess etc (but thats the easy part)

What is your advice in regards to [Chassis](https://github.com/netbox-community/devicetype-library/blob/master/device-types/Dell/PowerEdge-MX7000.yaml) and [IOM (example)](https://infohub.delltechnologies.com/en-us/l/dell-poweredge-mx-deployment-with-vmware-cloud-foundation-deployment-guide/dell-networking-mx9116n-fabric-switching-engine/) modeling.

My issue here is that the chassis is modeled with device bays for everything. The front chassis slots can be used for plugging servers (but also storage) blades.

The back slots are for power (correctly modeled), Networking fabric (in 2 redundant groups) and storage fabric in a redundant group. Total is 6 IOM's

The before mentioned device type models this as just device bays.

The IOMs usually have 25Gb virtual ports (for internal usage in the chassis) and external ports. Then to boot they are usually qsfp's or whatever (e.g. module bays). To top that ports can be broken out to multiple types :). And even more some ports can be dual purpose (e.g. interconnect, networking, fiber channel, etc).

And the fun does not stop here. Depending how you configure the chassis and the fabric group interfaces can show in the inserted blades.

So any advice would be appreciated.

**Comment by harryajc** [comment]
Hi @zeridon
How is should be done is that the blades on the front should be modeled as `device-bays`, the fabric/network modules in the back and PSUs should be modeled under `module-bays`. Then you will then need to create all of the required modules for the chassis.

**Reply by zeridon** [reply]
That is clear. More to the point i am seeking advice mainly how to model the IOM cards. As they are complex beasts that could be devices but could be also a module to plug in the chassis

**Reply by harryajc** [reply]
I would say as they are effectively a network expansion card they class as a module.
